File: tests/factories/review.py
import factory
import random

from salsa import models

from .base import ModelFactory
from .user_account import UserAccountFactory
from .product import ProductFactory
from .purchase import PurchaseFactory


class ReviewFactory(ModelFactory):
    name = factory.Faker('text')
    description = factory.Faker('text')
    numstars = random.randint(1,6)

    user_account = factory.SubFactory(UserAccountFactory)
    product = factory.SubFactory(ProductFactory)
    purchase = factory.SubFactory(PurchaseFactory)

    class Meta:
        model = models.Review

File: flutter/helistrong/lib/models/review.dart
class Review {
  final String id;
  final String name;
  final String description;
  final double numStars;
  final String purchaseId;
  final String userId;
  final DateTime createdAt;
  final DateTime updatedAt;

  Review(this.id, this.name, this.description, this.numStars, this.purchaseId, this.userId, this.createdAt, this.updatedAt);
}
